genesyscloud/responsemanagement_responseasset/genesyscloud_responsemanagement_responseasset_proxy.go

```diff
@@ -0,0 +1,110 @@
+	api := platformclientv2.NewResponseManagementApiWithConfig(clientConfig)
+	return &responsemanagementResponseassetProxy{
+		clientConfig:                       clientConfig,
+		responseManagementApi:              api,
+		createRespManagementRespAssetAttr:  createRespManagementRespAssetFn,
+		updateRespManagementRespAssetAttr:  updateRespManagementRespAssetFn,
+		getRespManagementRespAssetByIdAttr: getRespManagementRespAssetByIdFn,
+		deleteRespManagementRespAssetAttr:  deleteRespManagementRespAssetFn,
+	}
+}
+
+// getRespManagementRespAssetProxy acts as a singleton to for the internalProxy.  It also ensures
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getRespManagementRespAssetProxy(clientConfig *platformclientv2.Configuration) *responsemanagementResponseassetProxy {
+	if internalProxy == nil {
+		internalProxy = newRespManagementRespAssetProxy(clientConfig)
+	}
+	return internalProxy
+}
+
+// createRespManagementRespAsset creates a Genesys Cloud responsemanagement responseasset by Id
+func (p *responsemanagementResponseassetProxy) createRespManagementRespAsset(ctx context.Context, respAsset *platformclientv2.Createresponseassetrequest) (*platformclientv2.Createresponseassetresponse, int, error) {
+	return p.createRespManagementRespAssetAttr(ctx, p, respAsset)
+}
+
+// updateRespManagementRespAsset updates a Genesys Cloud responsemanagement responseasset by Id
+func (p *responsemanagementResponseassetProxy) updateRespManagementRespAsset(ctx context.Context, id string, respAsset *platformclientv2.Responseassetrequest) (*platformclientv2.Responseasset, int, error) {
+	return p.updateRespManagementRespAssetAttr(ctx, p, id, respAsset)
+}
+
+// getRespManagementRespAssetById returns a single Genesys Cloud responsemanagement responseasset by Id
+func (p *responsemanagementResponseassetProxy) getRespManagementRespAssetById(ctx context.Context, id string) (*platformclientv2.Responseasset, *platformclientv2.APIResponse, error) {
+	return p.getRespManagementRespAssetByIdAttr(ctx, p, id)
+}
+
+// deleteRespManagementRespAsset deletes a Genesys Cloud responsemanagement responseasset by Id
+func (p *responsemanagementResponseassetProxy) deleteRespManagementRespAsset(ctx context.Context, id string) (response *platformclientv2.APIResponse, err error) {
+	return p.deleteRespManagementRespAssetAttr(ctx, p, id)
+}
+
+// createRespManagementRespAssetFn is an implementation of the function to create a Genesys Cloud responsemanagement responseasset
+func createRespManagementRespAssetFn(ctx context.Context, p *responsemanagementResponseassetProxy, respAsset *platformclientv2.Createresponseassetrequest) (*platformclientv2.Createresponseassetresponse, int, error) {
+	postResponseData, resp, err := p.responseManagementApi.PostResponsemanagementResponseassetsUploads(*respAsset)
+	if err != nil {
+		return nil, 0, fmt.Errorf("Failed to upload response asset: %v", err)
```

Comment:
You need to return `resp.StatusCode` here. Retry logic is based on the status code